lib/colorado_lottery.rb

```diff
@@ -0,0 +1,52 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants= Hash.new { |hash, key| hash[key] = [] }
+  end
+
+  def interested_and_18?(contestant, game)
+    interested = contestant.game_interests.any? { |interested| interested == game.name }
+    interested && contestant.age >= 18
+  end
+
+  def can_register?(contestant, game)
+    in_state_or_national = (!contestant.out_of_state? || game.national_drawing)
+    interested_and_18?(contestant, game) && in_state_or_national
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game)
+      @registered_contestants[game.name] = [] if !@registered_contestants[game.name]
+      @registered_contestants[game.name] << contestant
+    end
+  end
+
+  def eligible_contestants(game)
+    @registered_contestants[game.name].find_all do |contestant|
+      contestant.spending_money >= game.cost
+    end
+  end
+
+  def charge_contestants(game)
+    @registered_contestants[game.name].each do |contestant|
+      if eligible_contestants(game).any?{|eligible| eligible == contestant}
```

Comment:
you could avoid this conditional by just using `eligible_contestants` as your collection instead of `@registered_contestants`-- then you know that you're only charging eligible contestants to start off with. 

```
def charge_contestants(game)
    current_contestants[game] = []

    eligible_contestants(game).each do |contestant|
      current_contestants[game] << contestant.full_name
      contestant.spending_money -= game.cost
    end
end
```